airbyte-integrations/connectors/destination-weaviate/destination_weaviate/client.py

```diff
@@ -0,0 +1,97 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import uuid
+import logging
+import json
+from typing import Any, Mapping
+
+import weaviate
+from .utils import generate_id, parse_id_schema, parse_vectors, stream_to_class_name
+
+
+class Client:
+    def __init__(self, config: Mapping[str, Any], schema: Mapping[str, str]):
+        self.client = self.get_weaviate_client(config)
+        self.config = config
+        self.batch_size = int(config.get("batch_size", 100))
+        self.schema = schema
+        self.vectors = parse_vectors(config.get("vectors"))
+        self.id_schema = parse_id_schema(config.get("id_schema"))
+
+    def queue_write_operation(self, stream_name: str, record: Mapping):
+        # TODO need to handle case where original DB ID is not a UUID
+        record_id = ""
+        if self.id_schema.get(stream_name, "") in record:
+            id_field_name = self.id_schema.get(stream_name, "")
+            record_id = generate_id(record.get(id_field_name))
+            del record[id_field_name]
```

Comment:
My IDE gives me the following warning on this line
```
Class 'Mapping' does not define '__delitem__', so the '[]' operator cannot be used on its instances
```
is this something to be concerned about?

Reply:
intellij with python plugin doesn't throw such an error so I don't think that's something to be concerned of. What IDE are you using?

Reply:
I'm using the same setup. can you please double check this? :)

Reply:
Same. Can you please double check this? :)

Reply:
I changed it to MutableMapping: https://docs.python.org/3/library/collections.abc.html#collections.abc.MutableMapping

That should solve the IDE warning. Can you verify?

Reply:
Yes, the warning is no longer present. And I guess it makes sense what the IDE was warning, if there is a Mapping and MutableMapping class you shouldn't be able to remove entries from an immutable data structure.